src/uu/printf/src/tokenize/num_format/formatters/cninetyninehexfloatf.rs

```diff
@@ -43,27 +43,25 @@ impl Formatter for CninetyNineHexFloatf {
 // c99 hex has unique requirements of all floating point subs in pretty much every part of building a primitive, from prefix and suffix to need for base conversion (in all other cases if you don't have decimal you must have decimal, here it's the other way around)
 
 // on the todo list is to have a trait for get_primitive that is implemented by each float formatter and can override a default. when that happens we can take the parts of get_primitive_dec specific to dec and spin them out to their own functions that can be overridden.
-#[allow(unused_variables)]
-#[allow(unused_assignments)]
 fn get_primitive_hex(
     inprefix: &InPrefix,
-    str_in: &str,
-    analysis: &FloatAnalysis,
-    last_dec_place: usize,
+    _str_in: &str,
+    _analysis: &FloatAnalysis,
+    _last_dec_place: usize,
     capitalized: bool,
 ) -> FormatPrimitive {
     let prefix = Some(String::from(if inprefix.sign == -1 { "-0x" } else { "0x" }));
 
     // assign the digits before and after the decimal points
     // to separate slices. If no digits after decimal point,
     // assign 0
-    let (mut first_segment_raw, second_segment_raw) = match analysis.decimal_pos {
-        Some(pos) => (&str_in[..pos], &str_in[pos + 1..]),
-        None => (str_in, "0"),
-    };
-    if first_segment_raw.is_empty() {
-        first_segment_raw = "0";
-    }
+    //let (mut first_segment_raw, second_segment_raw) = match analysis.decimal_pos {
```

Comment:
why comment this part?

Reply:
Dead code
Commented so I could remove warnings about unused/unread variables

Reply:
ok, just remove it then :)

We still have access to the history in git if we need to